cmd/babylond/cmd/root.go

```diff
@@ -308,9 +310,24 @@ func newApp(logger log.Logger, db dbm.DB, traceStore io.Writer, appOpts serverty
 	// auto migrate when build tag is set to "e2e_upgrade"
 	automigrate_e2e_upgrade(logger, homeDir)
 
-	blsSigner, err := appsigner.InitBlsSigner(homeDir)
-	if err != nil {
-		panic(fmt.Errorf("failed to initialize priv signer: %w", err))
+	// Try to load existing BLS signer first
+	blsSigner := appsigner.LoadBlsSignerIfExists(homeDir)
+
+	// If no existing signer, create new one with password based on flags
+	if blsSigner == nil {
+		password := ""
+		if !cast.ToBool(appOpts.Get(flagNoBlsPassword)) {
+			password = cast.ToString(appOpts.Get(flagBlsPassword))
+			if password == "" {
+				password = appsigner.NewBlsPassword()
+			}
+		}
+
+		var err error
+		blsSigner, err = appsigner.CreateBlsSigner(homeDir, password)
+		if err != nil {
+			panic(fmt.Errorf("failed to create new BLS signer: %w", err))
+		}
 	}
```

Comment:
I think we probably can wrap this chunk into `LoadOrGenBlsKey` like [here](https://github.com/cosmos/cosmos-sdk/blob/4e74a30430c8c2bcb9f760d7df287b2388a48702/server/start.go#L381)